### ISSUE TITLE
common scramblings strategy

### ISSUE THREAD

**Issue by joshcrews** [TheTumboliaSociety/reanonymizer#1]
thinking out loud about strategies:  what about something that did innocent misspelling, alternative spellings, or rephrasings every 50ish words?  Maybe the 50-ish part is randomized itself.  It would common things like swapping out "can't" for "cant"

That would keep more comment original message faithfulness than foreign language translation, but probably be sufficient to spoil mediocre de-anonymization bots.  Serious bots, especially ones that are working closely with analysts looking at specific cases would be expected to defeat this.

The whole thing runs as a Chrome plugin.

**Comment by leurfete** [comment]
I think that has nice potential. Is there a repo we could link?

**Comment by leurfete** [comment]
This could be upgraded over time with a thesaurus and replacement rules for various grammatical constructs (e.g. can't => cannot | can not). By keeping the replacements random we might confuse things a bit. 100% worth a try.

**Comment by joshcrews** [comment]
no, i'm just trying to start a discussion.  I have no experience in this field.

I just thought of another scramble: toggle capitalization on words that are normally capitalized but aren't strange if not.  Another scramble: replace abbreviations with full words like YMMV, and converse replace phrases with common abbreviations

**Comment by leurfete** [comment]
Makes sense. I'll start a repo here then. Should be straight forward to get started. :100: idea.

**Comment by joshcrews** [comment]
now that we are talking about it, i'm concerned that in just toggling easily replaceable things, the "voice" of the writer still shines through too distinctly, so I'm still thinking if this is a good strategy.  if the "voice" shines through, a fingerprint will be discernable, like whatever the writing analysis equivalent of Shazam would be able to break it.

**Comment by leurfete** [comment]
We would need to mangle things very significantly to make this work. I think if we also restrict the comment text to simple English (a corpus of just ~2000 words) it may be viable.

**Comment by leurfete** [comment]
All in all, the approach might be to.

1.) Replace / deny all works which aren't "Basic English" https://en.wikipedia.org/wiki/Basic_English
2.) Expand / replace as many grammatical structures and common phrases as possible (e.g. "one does not simply" => "one does not", "that just ain't so" => "that isn't so", "here a while back" => "some time ago")
